sklearn/inspection/_partial_dependence.py

```diff
@@ -321,6 +370,11 @@ def partial_dependence(
             `kind='average'` will be the new default. It is intended to migrate
             from the ndarray output to :class:`~sklearn.utils.Bunch` output.
 
+    custom_grid: dict
+        A dictionary mapping an element of `features` to an array of values where
+        the partial dependence should be calculated for that feature. The length
+        of `custom_grid` must match the length of `features`.
```

Comment:
I think the requirement that `len(custom_grid) == len(features)` is not a great user experience. 

I was thinking of extending `_grid_from_X` to support `custom_grid`, and if `custom_grid` is provide for a given feature, then do minimal validation and add it to `values`.